projects/rocprofiler-sdk/source/lib/rocprofiler-sdk/counters/evaluate_ast.cpp

```diff
@@ -468,18 +492,18 @@ EvaluateAST::set_dimensions()
         case MULTIPLY_NODE:
         case DIVIDE_NODE:
         {
-            auto first  = _children[0].set_dimensions();
-            auto second = _children[1].set_dimensions();
+            auto first  = _children[0].set_dimensions(agent_id);
```

Comment:
Is there at `.at(size_t)` that can used in lieu of `operator[](size_t)`? Or is this intentionally allocating if it doesn't exist?